### DISCUSSION TITLE
Build fails when we import path or fs module in api router inside app

### DISCUSSION THREAD

**Discussion by mighty98** [vercel/next.js#51479] [App Router]
Hi 
I am building an application using nextjs13.4.
I have a client component which needs to read data from the local folder (posts)
I created an api under app which is supposed to read the files and response with a json but the build fails saying module not found for 'fs' and 'path'
![image](https://github.com/vercel/next.js/assets/32213921/40130758-7fa8-400f-960d-623aefc86092)
```
Module not found: Can't resolve 'fs'
  1 | import path from 'path';
> 2 | import fs from 'fs';
  3 | import matter from 'gray-matter';
  4 | import html from 'remark-html';
  5 | import { remark } from 'remark';

```
Can someone please help me

**Comment by icyJoseph** [comment]
I guess you are using:

```ts
export const runtime = "edge";
```

In the Edge runtime, you have a V8 engine, that's not entirely Node API compatible. 

- https://vercel.com/docs/concepts/functions/edge-functions/edge-runtime#unsupported-apis:

> Some Node.js APIs other than the ones listed above are not supported. For example, you can't read or write to the filesystem

**Reply by mighty98** [reply]
@icyJoseph Thank you.. Is there any workarounds for using these in edge runtime?
Or is there a way we can have SSG and SSR together
Actually i am trying to host my nextjs app in cloudfare which supports either of these but not both. So at build time is it possible to have all these static pages to be generated and use it along with SSR?

**Reply by icyJoseph** [reply]
Could you describe what you want to do? without saying SSG/SSR? What's Cloudflare not supporting?

**Reply by mighty98** [reply]
Yeah sorry about vague description

Here is what i am trying to do
We have an application where we have a home page with images and text (basic html), a contact us page where we take user details and add to mongo db and a blog page where we read from .md file and display posts
We are using nextjs 13.4 version

Before introducing the blog section we were hosting in cloudfare pages our nextjs project and created a mongodb atlas worker in cloudfare
im using the latest /app folder
im using actions.js for connecting to mongodb 

Until now all were good
For blog i wanted to keep all the blog posts in local folder as .md files and then create html from these using gray-matter, remark etc

Any suggestions how i can use this?

Since i need to connect to mongodb i think i cannot host this in cloudfare as static export but has to use edge env
But in edge env i cannot find a way to make this blog section work?

**Reply by icyJoseph** [reply]
Aha, but can't you just use @next/mdx? https://nextjs.org/docs/app/building-your-application/configuring/mdx#nextmdx 

It has a downside though... for each post you'd have `MyPost.mdx` and a "glue" file, `page.tsx` that re-exports the markdown.

- https://nextjs.org/docs/app/building-your-application/configuring/mdx#nextmdx
- https://nextjs.org/docs/app/building-your-application/configuring/mdx#frontmatter

**Comment by mighty98** [comment]
Oh thanks for these guides @icyJoseph So this works with edge runtime?
https://blog.logrocket.com/create-next-js-mdx-blog/
I was going through this article and i see we need to use 'fs' module to get all mdx file list and read them

**Reply by icyJoseph** [reply]
Mmm no no, I meant that you'd have to manually export each markdown file. No fs, no path. All should be done at build time.

**Reply by mighty98** [reply]
oh ok.. let me try that.. Thanks alot :)
